code/game/turfs/turf.dm

```diff
@@ -103,6 +107,27 @@
 /turf/proc/Initalize_Atmos(times_fired)
 	CALCULATE_ADJACENT_TURFS(src)
 
+/turf/examine(mob/user)
+	. += ..()
+	if(!damage_amount)
+		. += "<span class='notice'>It looks fully intact.</span>"
+	else
+		var/dam = round((turf_heat_resistance/turf_max_heat_resistance)* 100, 1)
+		. += "<span class='warning'>The remaining integrity is at [dam]%</span>"
+
+/turf/welder_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(user.a_intent == INTENT_GRAB)
+		if(turf_heat_resistance >= turf_max_heat_resistance)
+			to_chat(user, "<span class='warning'>[src] is already in good condition!</span>")
+			return
+		if(!I.tool_start_check(user, amount=0))
+			return
+		to_chat(user, "<span class='notice'>You begin repairing [src]...</span>")
+		if(I.use_tool(src, user, 4 SECONDS, volume=50))
+			turf_heat_resistance = turf_max_heat_resistance
+			to_chat(user, "<span class='notice'>You repair [src].</span>")
+
```

Comment:
If you don't `return TRUE` in this proc it will proceed with a normal attack/bash.

Reply:
`return TRUE` at the end?

Reply:
Whenever you want an attack not to happen. Do you want the person to bash the turf (if that was a valid attack) after the warning that it is already in good condition?